src/dsql/ExprNodes.cpp

```diff
@@ -12297,6 +12297,30 @@ DmlNode* SysFuncCallNode::parse(thread_db* tdbb, MemoryPool& pool, CompilerScrat
 
 	node->args = PAR_args(tdbb, csb);
 
+	if (name == "MAKE_DBKEY")
+	{
+		// Special handling for system function MAKE_DBKEY:
+		// convert constant relation name into ID at the parsing time
+
+		auto literal = nodeAs<LiteralNode>(node->args->items[0]);
+
+		if (literal && literal->litDesc.isText())
+		{
+			const MetaName relName = literal->getText();
+			const jrd_rel* const relation = MET_lookup_relation(tdbb, relName);
```

Comment:
Shouldn't also better register a dependency on the relation?

Reply:
The function itself makes no actions with relation. All it needs - know relation ID.
Thus, I don't think we need any kind of dependency here.

Reply:
I understand, but if one drops a relation directly referenced by name inside a given procedure, next time this procedure will not be loaded. Dependencies exists just to avoid that.

Reply:
I still see no need in dependency - it is too restrictive in this case. 
I'll better raise error at the execution phase, if relation was not found at parse time.

Reply:
In this function the relation name is a literal, not an identifier. Thus I agree a dependency is not needed, a runtime check looks enough.